graylog2-server/src/main/java/org/graylog/plugins/pipelineprocessor/rulebuilder/parser/ActionParser.java

```diff
@@ -47,32 +46,42 @@ String generateAction(RuleBuilderStep step, boolean generateSimulatorFields) {
             throw new IllegalArgumentException("Function " + step.function() + " not available as action for rule builder.");
         }
 
-        FunctionDescriptor<?> function = actions.get(step.function()).descriptor();
+        final RuleFragment ruleFragment = actions.get(step.function());
+        FunctionDescriptor<?> function = ruleFragment.descriptor();
+
         String syntax = "  ";
         if (Objects.nonNull(step.outputvariable())) {
-            if (function.returnType().equals(Void.class)) {
+            if (function.returnType().equals(Void.class)) { // cannot set output variables for functions returning void
                 throw new IllegalArgumentException("Function " + step.function() + " does not return a value.");
+            } else if (ruleFragment.isFunction()) { // only set output variables to result of functions
```

Comment:
maybe for debugging we should throw an error here when we have an output that is not the result of a function, since this shouldn't be possible or?